src/lib/swapper/types.ts

```diff
@@ -353,7 +353,10 @@ export interface TradeExecutionBase {
   on<T extends TradeExecutionEvent>(eventName: T, callback: TradeExecutionEventMap[T]): void
 
   exec?: (input: TradeExecutionInput) => Promise<{ cancelPolling: () => void } | void>
-  execWalletAgnostic?: (
-    input: TradeExecutionInput2,
+  execEvm?: (input: EvmTradeExecutionInput) => Promise<{ cancelPolling: () => void } | void>
+  execCow?: (input: CowTradeExecutionInput) => Promise<{ cancelPolling: () => void } | void>
```

Comment:
The terminology might be a bit confusing here - `execUtxo` and `execEvm` refers to execution implementation by chain (as inferred by the input type name as well) whereas `execCow` refers to a swapper specific implementation.
Perhaps some vernacular with `sign/message` is more appropriate here?

Reply:
Or alternatively, what about fitting Cow in `execEvm` umbrella? The only real discriminators here are:

- `type CowAccountMetadata = { from: string }` vs. `type EvmAccountMetadata = { from: string; nonce: string }` i.e lack of nonce for CoW
- `type EvmTradeExecutionProps = { signAndBroadcastTransaction:(transactionRequest: EvmTransactionRequest) => Promise<string>}` vs. `type CowTradeExecutionProps = { signMessage: (messageToSign: Uint8Array) => Promise<string>}` i.e `signMessage` vs. `signAndBroadcastTransaction` indicates we're dealing with a CoW trade

It seems to me like these would be relatively trivial to discriminate against and allow us keeping it all umder one `EVM` umbrella that can handle both messages signing, and actual Txs

Reply:
Not sure Id go for combining them, or perhaps we can discuss a bit. I'm considering use case of using types to document the implementation requirements for a given swapper. Cowswap has a very wierd implementation that is totally atypical so Ive been keeping it disjoint to the rest of the more typical ones. 

It kinda wouldn't make sense for developers (as in developers using our SDK) implementing cowswap have to implement `signAndBroadcastTransaction` and `nonce`, or for all other evm swappers to implement `signMessage` but then again making them optional to avoid this lands us back into cast hell, and requires knowing the internal workings of the swappers to know what swapper needs what dependency.

It might be workable to implement stubs in those cases, but that basically negates the usefulness of an interface because developers would need to know the internals of the swapper to know they can get away with that.

Reply:
One hundred percent agree the naming is very much less than ideal. I'm actually hoping the best naming and location for these will emerge as we implement the final swappers

Reply:
Worth noting here that the swapper type system has been aligned to chain namespace historically, but this forced pattern has been the root cause of our type system woes. I think the type system should be aligned to implementation with commonalities abstracted where possible, but also deliberately and explicitly broken out where misalignment is spotted.

In the case of cowswap, it's a sore thumb and it implementation really doesn't align to the evm pattern the others implement.